### ISSUE TITLE
When API throws an error, promise is never fulfilled

### ISSUE THREAD

**Issue by KODIKAS-NL** [mondaycom/monday-sdk-js#26]
I'm writing an embedded app and I noticed that if the API throws an error, this is not communicated back to app running in the iframe, and because of this, the promise is also never fulfilled.

This is the code calling the local api ->
https://github.com/mondaycom/monday-sdk-js/blob/8ce514b085b0ac213d2c4640e437594966d00211/src/client.js#L107
Only resolve is implemented.

This is code on the monday site actually calling the api:
<img width="329" alt="image" src="https://user-images.githubusercontent.com/58047787/84025313-9d21ae80-a98b-11ea-99fe-3be279143178.png">

If an error occurs it is logged to the console, but not communicated back to the app running in the iframe.

I can work around this temporarily with a timeout, but it would be nice to get the actual error.

**Comment by mitul69** [comment]
I have same use, 
I have try with await but and added try catch but never comes with catch as well. 

When there is some issue in api call it always break the application is there any solution??

**Comment by KODIKAS-NL** [comment]
> I have same use,
> I have try with await but and added try catch but never comes with catch as well.
> 
> When there is some issue in api call it always break the application is there any solution??

I work around it using a 30 second timeout, here is my Typescipt code:

```
private async callApi<T>(query: string, options?: { token?: string, variables?: { [key: string]: any } }): Promise<T> {
        return new Promise(async (resolve, reject) => {
            var running = true;
            setTimeout(() => {
                if (!running)
                    return
                reject("Timeout, an error probably occured")
                running = false
            }, 30000)

            try {
                var result = await this.mondayApi.callApi<T>(query, options)
                if (!running)
                    return
                resolve(result)
            } catch (e) {
                reject(e)
            }
            running = false
        })
    }
```

**Comment by mitul69** [comment]
@KODIKAS-NL Thank you for. 

But this is not good way to wait for 30 seconds. For me I am calling couple of queries I have change your login 10 seconds 
thank you

**Comment by KODIKAS-NL** [comment]
> @KODIKAS-NL Thank you for.
> 
> But this is not good way to wait for 30 seconds. For me I am calling couple of queries I have change your login 10 seconds
> thank you

Looks like a real fix is in the works: https://github.com/mondaycom/monday-sdk-js/tree/feat/vlad/propogate-api-errors

**Comment by vladmonday** [comment]
Solved in 0.1.1 version
